### PR TITLE
feat: atomic latency option added to Ansible

### PR DESCRIPTION
**Pull request by ktechmidas**
## Issue being fixed or feature implemented

To add "random" latency at the network level to the masternode servers through Ansible in order to test Platform under latency more similar to how mainnet operates. This will apply to devnet/testnet only

## What was done?
* Two roles have been added to ansible as the underlying Linux command is not atomic
  * The first is to add network latency if latency == 1 in the ansible vars
  * The second will set the latency to nothing if latency variable is 0 (This is so each command can run atomically)

This is achieved using qdiscs (Linux traffic control) and applies to all traffic to and from the masternode in question. Once you set latency to 1 it will apply to all masternodes

Note that right now we are simply using a shell command to achieve this as there is no underlying Ansible module which can add latency. This will be revisited later with an actual module...

## How Has This Been Tested?

Tested by bringing up my own devnet and applying the modules then attempting a simple ping.

* With latency = 0 (default)

ubuntu@masternode-4:~$ ping google.com
PING google.com (142.250.217.78) 56(84) bytes of data.
64 bytes from sea09s29-in-f14.1e100.net (142.250.217.78): icmp_seq=1 ttl=90 time=8.54 ms
64 bytes from sea09s29-in-f14.1e100.net (142.250.217.78): icmp_seq=2 ttl=90 time=8.53 ms

* With latency = 1

buntu@masternode-4:~$ ping google.com
PING google.com (142.250.217.78) 56(84) bytes of data.
64 bytes from sea09s29-in-f14.1e100.net (142.250.217.78): icmp_seq=1 ttl=90 time=108 ms
64 bytes from sea09s29-in-f14.1e100.net (142.250.217.78): icmp_seq=2 ttl=90 time=102 ms

## Breaking Changes

None, since it's off by default, with latency = 1 I haven't noticed any breaking changes yet on my devnet.

It will add an "empty" qdisc when no latency is applied

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by shumkov** [comment]
I wouldn't consider it as a breaking change since it's a new functionality so if you don't touch it then everything works the same as before.

File: ansible/deploy.yml [hunk 1]
```diff
@@ -214,6 +214,16 @@
       when: evo_services
     - role: elastic-beats
 
+
+# Set up fake latency on masternodes (if enabled)
+- hosts: masternodes
+  become: true
+  roles:
+    - role: add-fake-latency
+      when: latency == 1
```
Comment (shumkov) [inline ansible/deploy.yml:223]: It seems for other variables in the project we use booleans.

File: ansible/group_vars/all [hunk 2]
```diff
@@ -139,3 +139,9 @@ elastic_username: elastic
 elastic_password:
 kibana_encryptionkey:
 elastic_max_heap_size: 2g
+
+#Fake latency stuff
+
+latency: 0
```
Comment (shumkov) [inline ansible/group_vars/all:145]: What if call it more specific like `masternodes_network_latency`?

File: ansible/group_vars/all [hunk 3]
```diff
@@ -139,3 +139,9 @@ elastic_username: elastic
 elastic_password:
 kibana_encryptionkey:
 elastic_max_heap_size: 2g
+
+#Fake latency stuff
+
+masternode_network_latency: false
+latency_min: 10
+latency_max: 100
```
Comment (shumkov) [inline ansible/group_vars/all:147]:
```suggestion
masternode_network_latency_min: 10
masternode_network_latency_max: 100
```
latency_max: 100

**Comment by shumkov** [comment]
As we discussed the latency thing affects deployment speed too. Is it possible to disable latency before we do deploy and enable it back as the last step?

File: ansible/deploy.yml [hunk 4]
```diff
@@ -2,217 +2,234 @@
 
 # Bootstrap
 
-- hosts: all
```
Comment (shumkov) [inline ansible/deploy.yml:5]: It seems you commited something wrong

**Review by shumkov — CHANGES_REQUESTED** [review]

File: ansible/deploy.yml (hunk 4 above)
Comment (ktechmidas) [reply ansible/deploy.yml:5]: Correct, I was testing... fixed it :|

**Review by shumkov — APPROVED** [review]
LGTM 👍

**Pull request merged**